### PR TITLE
Support resource limits in C, C++ API

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: libshaderc/include/shaderc/shaderc.h [hunk 1]
```diff
@@ -255,6 +342,10 @@ void shaderc_compile_options_set_target_env(shaderc_compile_options_t options,
 void shaderc_compile_options_set_warnings_as_errors(
     shaderc_compile_options_t options);
 
+// Sets a resource limit.
+void shaderc_compile_options_set_limit(
+    shaderc_compile_options_t options, shaderc_limit limit, int value);
```
Comment (antiagainst) [inline libshaderc/include/shaderc/shaderc.h]: I'm not particular clear about this part tho: should we use an explicitly sized integer type here?

File: libshaderc/include/shaderc/shaderc.h [hunk 2]
```diff
@@ -86,6 +86,93 @@ typedef enum {
   shaderc_optimization_level_size,  // optimize towards reducing code size
 } shaderc_optimization_level;
 
+// Resource limits.
+typedef enum {
```
Comment (AWoloszyn) [inline libshaderc/include/shaderc/shaderc.h]: It is unfortunate to have to specify everything twice, once in the .inl file and once here, but I understand the desire to keep the header as clean as possible.

File: libshaderc_util/include/libshaderc_util/resources.inc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These are the resource limits in a glslang::TBuiltInResource.
+// The first field is the string name to be used in a configuration setting.
+// The second field is the fieldname in TBuiltInResource.
```
Comment (antiagainst) [inline libshaderc_util/include/libshaderc_util/resources.inc]: But I see three fields...

File: libshaderc/include/shaderc/shaderc.h (hunk 2 above)
Comment (dneto0) [inline libshaderc/include/shaderc/shaderc.h]: Yeah.  This is like circularity we've seen before:
* the public header is in libshaderc/shaderc.h
* but libshaderc_util code has to talk about those enums to translate down into Glslang.
So we need two mirrored enum sets.  The best I can do is the single resources.inc file that makes the association.

File: libshaderc/include/shaderc/shaderc.h (hunk 1 above)
Comment (dneto0) [inline libshaderc/include/shaderc/shaderc.h]: Glslang uses int.  If we went explicitly sized then there could potentially be an inconvenient  mismatch someday.

File: libshaderc_util/include/libshaderc_util/resources.inc (hunk 3 above)
Comment (dneto0) [inline libshaderc_util/include/libshaderc_util/resources.inc]: good catch.  :-)

**Comment by dneto0** [comment]
Fixed the comment in resources.inc.
But I have a legitimate gcc failure to fix.

**Comment by dneto0** [comment]
Squashed

**Pull request merged**